packages/system/src/xydrag/XYDrag.ts

```diff
@@ -87,14 +94,13 @@ export type DragUpdateParams = {
   nodeClickDistance?: number;
 };
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function XYDrag<OnNodeDrag extends (e: any, nodes: any, node: any) => void | undefined>({
+export function XYDrag<NodeType extends NodeBase = NodeBase>({
```

Comment:
It would be better to pass EdgeType here too, so that we can pass it to XYDragParams instead of having to type `getStoreItems` directly.